api/intents_api.py

```diff
@@ -22,21 +22,46 @@
 from framework import basehandlers
 from framework import cloud_tasks_helpers
 from framework import permissions
-from internals import processes
 from internals import stage_helpers
-from internals.core_enums import INTENT_STAGES_BY_STAGE_TYPE
+from internals.core_enums import (
+  INTENT_DRAFT_TYPES_BY_STAGE_TYPE,
+  FEATURE_TYPE_DEPRECATION_ID,
+  IntentDraftType
+)
 from internals.core_models import FeatureEntry
 from internals.review_models import Gate
-from pages.intentpreview import compute_subject_prefix
 import settings
 
 
+SUBJECT_PREFIXES = {
+  IntentDraftType.PROTOTYPE: 'Intent to Prototype',
+  IntentDraftType.DEVELOPER_TESTING: 'Ready for Developer Testing',
+  IntentDraftType.EXPERIMENT: 'Intent to Experiment',
+  IntentDraftType.EXTEND_EXPERIMENT: 'Intent to Extend Experiment',
+  IntentDraftType.PSA: 'Web-Facing Change PSA',
+  IntentDraftType.DEPRECATE: 'Intent to Deprecate and Remove',
+  IntentDraftType.SHIP: 'Intent to Ship'
+}
+
+
+def compute_subject_prefix(
+    feature_type: int, intent_type: IntentDraftType):
+  """Compute the subject line prefix for an intent email based on the feature and intent type."""
+  # Deprecation-specific intent names.
+  if feature_type == FEATURE_TYPE_DEPRECATION_ID:
```

Comment:
While the calling code in `do_get` and `do_post` has checks that should prevent an unknown type from ever reaching this function, it might be slightly safer to have it raise an exception instead of returning a default string. This would make it fail fast if an unexpected `intent_type` is ever passed in.

What do you think about changing this to something like:
```python
  # Get the prefix, or raise an error if the type is unknown.
  prefix = SUBJECT_PREFIXES.get(intent_type)
  if prefix is None:
    raise ValueError(f'Unknown IntentDraftType: {intent_type}')
  return prefix
```
This is a minor suggestion, and the current implementation is also reasonable.